parser/src/lexer.rs

```diff
@@ -77,45 +77,58 @@ impl Location {
     }
 }
 
+macro_rules! construct_keywords {
```

Comment:
I am no big fan of macros. To my opinion the use of the macro here makes the code less clear.